sdks/java/core/src/main/java/org/apache/beam/sdk/util/PathUtils.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Strings;
+import java.net.URI;
+
+/**
+ * Utility class for handling file paths.
+ */
+public class PathUtils {
+
+  private static final String URI_DELIMITER = "/";
+
+  /**
+   * Resolve multiple {@code others} against the {@code directory} sequentially.
```

Comment:
Phrasing cleanup:

Resolves multiple path fragments against the given directory sequentially.

Unlike URI#resolve, this function appends relative path fragments rather than replaces them, e.g. (give example).